src/shared_modules/content_manager/src/components/offlineDownloader.hpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Wazuh content manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * October 24, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _OFFLINE_DOWNLOADER_HPP
+#define _OFFLINE_DOWNLOADER_HPP
+
+#include "hashHelper.h"
+#include "json.hpp"
+#include "stringHelper.h"
+#include "updaterContext.hpp"
+#include "utils/chainOfResponsability.hpp"
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+/**
+ * @class OfflineDownloader
+ *
+ * @brief Class in charge of copying a file from the filesystem and update the context accordingly, as a step of a chain
+ * of responsibility.
+ *
+ */
+class OfflineDownloader final : public AbstractHandler<std::shared_ptr<UpdaterContext>>
+{
+private:
+    /**
+     * @brief Pushes the state of the current stage into the data field of the context.
+     *
+     * @param contextData Reference to the context data.
+     * @param status Status to be pushed.
+     */
+    void pushStageStatus(nlohmann::json& contextData, std::string status) const
+    {
+        auto statusObject = nlohmann::json::object();
+        statusObject["stage"] = "OfflineDownloader";
+        statusObject["status"] = std::move(status);
+
+        contextData.at("stageStatus").push_back(std::move(statusObject));
+    }
+
+    /**
+     * @brief Function to calculate the hash of a file.
+     *
+     * @param filepath Path to the file.
+     * @return std::string Digest vector.
+     */
+    std::string hashFile(const std::filesystem::path& filepath) const
+    {
+        if (std::ifstream inputFile(filepath, std::fstream::in); inputFile)
+        {
+            constexpr int BUFFER_SIZE {4096};
+            std::array<char, BUFFER_SIZE> buffer;
+
+            Utils::HashData hash;
+            while (inputFile.read(buffer.data(), buffer.size()))
+            {
+                // LCOV_EXCL_START
+                hash.update(buffer.data(), inputFile.gcount());
+                // LCOV_EXCL_STOP
+            }
+            hash.update(buffer.data(), inputFile.gcount());
+
+            return Utils::asciiToHex(hash.hash());
+        }
+
+        throw std::runtime_error {"Unable to open '" + filepath.string() + "' for hashing."};
+    };
+
+    /**
+     * @brief Downloads the requested local file and update the context accordingly.
+     *
+     * @note Despite the method name, there is no such download since the file is present in the filesystem.
+     *
+     * @param context Updater context.
+     */
+    void download(UpdaterContext& context) const
+    {
+        // Take the 'url' as the input file path.
+        auto url {context.spUpdaterBaseContext->configData.at("url").get_ref<const std::string&>()};
```

Comment:
I think it should be like this if we don't want to modify the URL ar configData:
```suggestion
        auto url {context.spUpdaterBaseContext->configData.at("url").get<std::string>()};
```
or like this if we want to modify it:
```suggestion
        auto& url {context.spUpdaterBaseContext->configData.at("url").get_ref<std::string&>()};
```

Reply:
Fixed! Just wanted to make a copy.